Common/src/QuasarThreadPool.cpp

```diff
@@ -48,39 +48,94 @@ ThreadPool::~ThreadPool ()
     while (!m_pendingJobs.empty())
     {
         ThreadPoolJob* job = m_pendingJobs.front();
-        m_pendingJobs.pop();
+        m_pendingJobs.pop_front();
         LOG(Log::WRN) << "Removing unfinished job: " << job->describe();
         delete job;
     }
 }
 
+/** This method finds the next job suitable to be dealt by the next available worker.
+ * It can finish with two potential outcomes:
+ * -- there is no suitable job to execute 
+ *      returns nullptr for the job, does not change the job list
+ * -- there is a suitable job to execute (either w/o a mutex or with a mutex that is free)
+ *      locks that specifix mutex (if applicable)
+ *      returns the job ptr and the lock
+ *      removes that job from the list
+ */ 
+ThreadPool::Duty ThreadPool::findSomeDuty ()
+{
+    for (auto iter = std::begin(m_pendingJobs); iter != std::end(m_pendingJobs); iter++)
```

Comment:
Consider replacing the for loop with std::find_if with a lambda. That should simplify the function.